research_project/greedy_triangulation.py

```diff
@@ -9,16 +9,32 @@
 
 from src.functions import poipairs_by_distance, new_edge_intersects
 
+def create_pois_groups(subgraph_percentages: list[float], pois: list[int]) -> list[int]:
+    pois_groups = []
+    pois_not_added = pois
+    for i in range (0, len(pois)):
+        group = random.sample(pois_not_added, int(len(pois) * subgraph_percentages[i]))
+        pois_groups.append(group)
+        pois_not_added.remove(group)
```

Comment:
This crashes because you're trying to remove a list of integers from a list of integers. E.g.

```py
>>> [0, 1, 2, 3, 4, 5].remove([1, 2, 3])
ValueError: list.remove(x): x not in list
```

```suggestion
        for poi in group:
            pois_not_added.remove(poi)
```